aws4/index.d.ts

```diff
@@ -30,4 +30,4 @@ export class RequestSigner {
     formatPath(): string;
 }
 
-export function sign(options?: any, credentials?: any): RequestSigner;
+export function sign(options?: any, credentials?: any): any;
```

Comment:
The `sign` function actually returns a modified version of the `options` parameter which can be either a string that is parsed into a url object or an object with any properties. The returned object also has arbitrary properties since it depends on how the url is parsed. `Object` could possibly be used instead of `any` but I'm not sure if TypeScript would handle that properly.